### PR TITLE
Add access rework doc, add domain doc

### PR DESCRIPTION
**Pull request by Firstyear**
Relates #990 doc changes to access controls to support service account and other improvements. 

- [ ] cargo fmt has been run
- [ ] cargo clippy has been run
- [ ] cargo test has been run and passes
- [ x ] book chapter included (if relevant)
- [ x ] design document included (if relevant)

### CODE REVIEW

File: kanidm_book/src/developers/designs/access_profiles_rework_2022.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+#### Service Account Admin
+
+The role would be called "sa\_admins" and would be responsible for top level management of service
+accounts, and delegating authority for service account administration to managing users.
+
+* Create service accounts
+* Delegate service account management to owners groups
+* Migrate service accounts to persons
+
+The service account admin is response to migrate service accounts to persons as it is "yielding"
```
Comment (yaleman) [inline kanidm_book/src/developers/designs/access_profiles_rework_2022.md:115]:
```suggestion
The service account admin is capable of migrating service accounts to persons as it is "yielding"
```

**Review by yaleman — CHANGES_REQUESTED** [review]
I like this 👍🏻

For the access control policies, would you see the ability to add OAuth RP's as always needing full system-admin, or would it be a capability that a custom admin role could have? It's something I could see as being a *relatively* low-level admin privilege compared to changing the domain name or replication configs.

We probably should pin down terminology to make these kinds of discussions easier 😄

**Comment by Firstyear** [comment]
> I like this 👍🏻
> 
> For the access control policies, would you see the ability to add OAuth RP's as always needing full system-admin, or would it be a capability that a custom admin role could have? It's something I could see as being a _relatively_ low-level admin privilege compared to changing the domain name or replication configs.

we could split it to a seperate admin permission for certain, but "by default" sysadmin would also have it. 

> 
> We probably should pin down terminology to make these kinds of discussions easier 😄

Yeeeaaaaaahhhhhhhhhhhhhhhhhhhhhhh.

**Comment by yaleman** [comment]
> > I like this 👍🏻
> > For the access control policies, would you see the ability to add OAuth RP's as always needing full system-admin, or would it be a capability that a custom admin role could have? It's something I could see as being a _relatively_ low-level admin privilege compared to changing the domain name or replication configs.
> 
> we could split it to a seperate admin permission for certain, but "by default" sysadmin would also have it.

Yeah, definitely retained by sysadmin, just making sure for my head-canon that it'll be *possible* with the new plan. 😄 

> > We probably should pin down terminology to make these kinds of discussions easier 😄
> 
> Yeeeaaaaaahhhhhhhhhhhhhhhhhhhhhhh.

random nonsense ideas below

| Term | Description | Examples (Vague, non-exhaustive) | 
| ---  | --- | --- |
| entity | something in the database | "uuid 12321-14512-12215", "domain_name = idm.example.com" |
| account | an entity with a uuid/username/authentication method | `william@idm.example.com`, `sa_supers3kret_h4ck3r@idm.crabz.ru` |
| capability | a verb which can apply to an entity | "edit password", "create outh2 RP", "see all properties of target" |
| target | what entity can the capability affect | "all persons", "all accounts", "some random group of things" |
| role | a named group of (capability => target) associations | "IDM admins" |

**Review by yaleman — APPROVED** [review]

**Comment by Firstyear** [comment]
> 
> Yeah, definitely retained by sysadmin, just making sure for my head-canon that it'll be _possible_ with the new plan. 😄
> 

Let me make it explicit in the doc then. 


> > Yeeeaaaaaahhhhhhhhhhhhhhhhhhhhhhh.
> 
> random nonsense ideas below

Should I add this to the design doc? Or in the rust doc of the access module? Perhaps there needs to be an access chapter in the book (eventually)

> 
> Term	Description	Examples (Vague, non-exhaustive)
> entity	something in the database	"uuid 12321-14512-12215", "domain_name = idm.example.com"
> account	an entity with a uuid/username/authentication method	`william@idm.example.com`, `sa_supers3kret_h4ck3r@idm.crabz.ru`
> capability	a verb which can apply to an entity	"edit password", "create outh2 RP", "see all properties of target"
> target	what entity can the capability affect	"all persons", "all accounts", "some random group of things"
> role	a named group of (capability => target) associations	"IDM admins"

**Comment by yaleman** [comment]
> > Yeah, definitely retained by sysadmin, just making sure for my head-canon that it'll be _possible_ with the new plan. 😄
> 
> Let me make it explicit in the doc then.
> 
> > > Yeeeaaaaaahhhhhhhhhhhhhhhhhhhhhhh.
> > 
> > 
> > random nonsense ideas below
> 
> Should I add this to the design doc? Or in the rust doc of the access module? Perhaps there needs to be an access chapter in the book (eventually)
> 

It should probably be a whole separate thing, maybe a glossary page which can include some of the domain terms?

**Comment by Firstyear** [comment]
Okay, added a glossary chapter and added some of your suggestions.

**Review by yaleman — APPROVED** [review]

**Pull request merged**